Circuit/Simulation/Simulation.cs

```diff
@@ -155,25 +155,14 @@ public Simulation(TransientSolution Solution)
         /// <param name="Output">Buffers to receive output samples.</param>
         public void Run(int N, IEnumerable<double[]> Input, IEnumerable<double[]> Output)
         {
-            if (process == null)
-                process = DefineProcess();
-
-            // Build parameter list for the processor.
-            object[] parameters = new object[2 + Input.Count() + Output.Count()];
-            int p = 0;
-
-            parameters[p++] = N;
-            parameters[p++] = n * TimeStep;
-            foreach (double[] i in Input)
-                parameters[p++] = i;
-            foreach (double[] i in Output)
-                parameters[p++] = i;
+            if (_process == null)
+                _process = DefineProcess();
 
             try
             {
                 try
                 {
-                    process.DynamicInvoke(parameters);
+                    _process(N, n*TimeStep, Input.ToArray(), Output.ToArray());
```

Comment:
I think these ToArray calls are also allocations. Rather than change the arguments of this to `double[][]`, which would be kind of a pain in some usages, I think we should change this to `AsArray` (add this helper if it doesn't exist), and tweak the callers to pass arrays *most* of the time when convenient (but keep the Concat when using the oscilloscope rather than make a mess trying to populate an array manually, which is going to do lots of allocations anyways).

Feel free to leave this to a follow-up change, I'd be happy to do this.

Reply:
(To clarify, here's what I meant by AsArray: https://github.com/dsharlet/ComputerAlgebra/blob/master/ComputerAlgebra/Utils/EnumerableExtensions.cs#L28-L40)